RETAIL-PLAN.md

```diff
@@ -0,0 +1,58 @@
+
+#### New Feature in WoW Voiceover Addon
+
+- Add a new feature to the WoW Voiceover addon that enables users to opt-in to collect quest/gossip text data during gameplay
+- Create a lightweight companion executable that runs in the background during gameplay to upload the scraped text data to a central server
+- The data should include:
+
+| Column         | Description                                                |
+|----------------|------------------------------------------------------------|
+| source         | Type of interaction ('accept', 'progress', 'complete', or 'gossip') |
+| quest          | Quest ID or empty string if it's a gossip interaction      |
+| text           | text content of the interaction |
+| race           | Race of the NPC involved in the interaction                 |
+| sex            | Gender of the NPC involved in the interaction               |
+| npc_name           | Name of the NPC involved in the interaction                 |
+| npc_type           | Type of the NPC involved ('creature', 'gameobject', or 'item') |
+| npc_id             | Creature/gameobject/item ID of the NPC involved in the interaction |
+| player_race    | Race of the player involved in the interaction              |
+| player_gender  | Gender of the player involved in the interaction            |
+| player_name  | Name of the player involved in the interaction            |
+
+- Race and Sex will probably not be possible to collect directly from the game addon for recent/current expansions.
+
+#### API and Database Development
+
+- Develop a robust API and database to store and manage the collected text data
+- The API should provide endpoints for data retrieval, updates, and submission of new data from the companion executable
+- Seed the database with initial data from WoWhead and other sources, and continually update it with scraped data from users' gameplay
+
+#### NPC Race and Sex Matching
+
+- Utilize projects like [wow-listfile](https://github.com/wowdev/wow-listfile) to regain filenames for each NPC, as model names are not available in newer expansions
+- Implement an algorithm or use an LLM (e.g., GPT-4) to make educated guesses of race and sex based on available information, such as filenames or sound names
+- Continuously refine and improve the race and sex matching algorithm as more data is collected and analyzed
+
+### Converting uploaded text to template text
+
+The API should be capable of handling variations in the text based on gender, name, and race differences. To achieve this, the API collects those pieces of information and post processes the uploaded text. Name should be easy to do a simple subsitution along with race. Gender typically has completely different dialog. Through a combination of subsitution then diffing user's submissions for the same piece of content, we should be able to make a good estimate as to what the template text is. The most up to date guess on the template text should be stored in the DB.
```

Comment:
We can obtain the exact original texts for quest's accept text from quest cache, but not for the progress and complete/reward texts.

We can also obtain the exact original texts for gossip from npccache.wdb cache (which reference BroadcastText.db2, which is bruteforceable, but I don't know how), but without explicit links to which npc does that text belong to. So connecting those texts to gossips scraped by the addon will still involve guesswork.

Reply:
Given how much added complexity this would be, and that it doesnt really "solve" the issue of collection, I'd like to start with the basic approach and not use the game's cache.

Ill add this information to the doc though.

Reply:
added